test/azure/rules.spec.ts

```diff
@@ -1,33 +1,42 @@
 import {assert} from 'chai';
-import {Rule, ResourceGraphRule} from '../../src/rules';
+import {
+  ResourceGraphRule,
+  ResourceGraphExecutor,
+  ResourceGraphTarget,
+  RuleType,
+} from '../../src/rules';
 import {subscriptionId} from '.';
 
 describe('Resource Graph Rule', function () {
-  this.slow(5000);
-  this.timeout(8000);
+  this.slow(6000);
+  this.timeout(10000);
 
   it('can execute a resource graph rule and return a scan result', async () => {
     const query =
       "Resources | where type =~ 'Microsoft.Compute/virtualMachines2'";
-    const name = 'Dummy Rule';
+    const name = 'test-rule';
     const description = 'Intentional bad query';
-    const rule: Rule = {
+    const rule: ResourceGraphRule = {
       name,
       query,
       description,
-      type: 'resourceGraph',
+      type: RuleType.ResourceGraph,
     };
-    const results = await ResourceGraphRule.execute([rule], subscriptionId);
+    const target: ResourceGraphTarget = {
+      type: RuleType.ResourceGraph,
+      subscriptionId,
+    };
+    const results = await ResourceGraphExecutor.execute([rule], target);
     for (const result of results) {
       assert.equal(rule.name, result.ruleName);
       assert.equal(rule.description, result.description);
       assert.containsAllKeys(result, [
         'ruleName',
         'description',
         'total',
-        'ids',
+        'resourceIds',
       ]);
-      assert.equal(result.total, 0);
+      assert.isAtLeast(result.total, 0);
```

Comment:
Can you confirm this is intended? If we run a query that returns 0 results, we should do one of two things:

* return 0 results
* throw an error (if the call to Resource Graph throws)

If we expect zero and get back 1, I don't think we'd want this test to pass

Reply:
This should be equal and not isAtLeast. I'm not sure why I changed that. This rule should always return 0 resources